Winleafs.Wpf/Api/Layouts/PanelLayout.cs

```diff
@@ -88,39 +88,64 @@ private void ConstructPanelsAsTriangles()
         }
 
         /// <summary>
-        /// Draws an equilateral triangle from the given center point and rotation. Also applies the global rotation
+        /// Draws an equilateral polygon from the given center point and rotation. Also applies the global rotation
         /// </summary>
-        private void CreateTriangle(double x, double y, double rotation, int panelId, Transform globalRotationTransform)
+        private void CreatePolygon(double x, double y, double rotation, int panelId,
+            Transform globalRotationTransform, int shapeType)
         {
-            //First assume that we draw the triangle facing up:
-            //     A
-            //    /\
-            //   /  \
-            //  /____\
-            // B      C
-
-            var A = new Point(x, y - ((Math.Sqrt(3) / 3) * _defaultTriangleSize));
-            var B = new Point(x - (_defaultTriangleSize / 2), y + ((Math.Sqrt(3) / 6) * _defaultTriangleSize));
-            var C = new Point(x + (_defaultTriangleSize / 2), y + ((Math.Sqrt(3) / 6) * _defaultTriangleSize));
-
             var rotateTransform = new RotateTransform(rotation, x, y);
 
             //Apply transformation and add points to polygon
-            var triangle = new Polygon();
-            triangle.Points.Add(globalRotationTransform.Transform(rotateTransform.Transform(A)));
-            triangle.Points.Add(globalRotationTransform.Transform(rotateTransform.Transform(B)));
-            triangle.Points.Add(globalRotationTransform.Transform(rotateTransform.Transform(C)));
+            var polygon = new Polygon();
+
+            switch (shapeType)
```

Comment:
Why not make shapeType an enum? Since the API gives back an int, its perfectly castable to an enum